### ISSUE TITLE
Why does this library exist?

### ISSUE THREAD

**Issue by eldoy** [developit/vhtml#38]
Why does this library exist when you can just use template strings?

```js
function myComponent(props) {
  return `<div>${props.name}</div>`
}
```

**Comment by Mike-Logit** [comment]
I agree. It would be good to explain in the README if there are reasons I'm unaware of. I'm using typescript files to convert to individual javascript files for an MVC app with no React and wanted to use this plugin in my gulp/webpack setup, but then I thought to just use template strings instead.

**Comment by zuisong** [comment]
For me, one reason is that template strings lack strong type.

For example:

```jsx
function myComponent(props) {
  return `<div>${props.name}</di>`  // <-- Here is a spelling mistake, but it is not easily noticeable.
}

```
jsx

```jsx
function myComponent(props) {
  return <div>{props.name}</di>  // <-- When using jsx , it will raise an error.
}

```

**Comment by eldoy** [comment]
@zuisong Some kind of linter in your editor can tell you that, no need for JSX or Babel or any other garbage like that.

**Comment by catdevnull** [comment]
it sanitizes text.

**Comment by jcubic** [comment]
You can have components and create really complex HTML with it. With template literal you only can create simple examples.

**Comment by jcubic** [comment]
It's also great for debugging JSX libraries (if it would work correctly).

**Comment by devingfx** [comment]
You use JSX syntax and helpers that you can't use with template literals...
Like binding event handlers, or attributes as object.
This is a matter of choice, sugar vs vanilla...

You can not say that this library is useless! It permit to use JSX syntax but not along with p?react or any other Rube Goldberg machine...
JSX became used everywhere nowadays (some newbies front devs don't know that HTML in js is not a standard 😂)

Note this vhtml library can also be used with template literals along with [htm](https://github.com/developit/htm) that kindof brings some JSX shothands to template literals ;)
